poetry/repositories/pypi_repository.py

```diff
@@ -421,11 +431,74 @@ def _get_info_from_urls(
                 return info
 
             if platform_specific_wheels and "sdist" not in urls:
-                # Pick the first wheel available and hope for the best
-                return self._get_info_from_wheel(platform_specific_wheels[0])
+                # Attempt to select the best platform-specific wheel
+                best_wheel = self._pick_platform_specific_wheel(
+                    platform_specific_wheels
+                )
+                return self._get_info_from_wheel(best_wheel)
 
         return self._get_info_from_sdist(urls["sdist"][0])
 
+    def get_sys_info(self):  # type: () -> Dict[str, str]
```

Comment:
could sys_info be a property on class `PyPiRepository`, or `Repository`? Given that it may be used elsewhere, this could prevent a few calls (not that they are very expensive).

Reply:
@KeltonKarboviak, thanks for the input. I reviewed the code base and I think that setting `sys_info` as a class property will actually be less performant. As far as I can tell, `PyPiRepository` is always initialized when used (the class and properties aren't reused), which means that if `sys_info` is a class property, then those functions will be called for each class instance/repository. However, the local hardware information is only needed for a subset of python repositories that otherwise can't be identified. When the hardware information is needed, it is only called once before looping through the potential wheel file names

Let me know if there is a something I overlooked. I already staged code changes, so this would be easy to implement

Reply:
@KyleKing I was thinking in the lines of this, which would only be called when cls().sys_info is called.

```python
@property
def sys_info(self):  # type: () -> Dict[str, str]
    # Return system information. Can be overridden for testing
    if self._sys_info:
        return self._sys_info
    self._sys_info = {
        "plat": platform.system().lower(),
        "is32bit": sys.maxsize <= 2 ** 32,
        "imp_name": sys.implementation.name,
        "pyver": platform.python_version_tuple(),
    }
    return self._sys_info
```

You would need to update 

https://github.com/sdispater/poetry/blob/41b321f990bd66e3053f76585f75188e629ec890/tests/repositories/test_pypi_repository.py#L53

https://github.com/sdispater/poetry/blob/41b321f990bd66e3053f76585f75188e629ec890/poetry/repositories/pypi_repository.py#L454

Reply:
Oh you're right, thanks!